.github/workflows/ci.yml

```diff
@@ -206,6 +186,40 @@ jobs:
         run: |
           node ./scripts/assert-dir-git-clean.js "lint-ci check-compat-data"
 
+  test-coverage:
+    name: Test on Node.js Latest (with coverage)
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Use Node.js latest
+        uses: actions/setup-node@v4
+        with:
+          node-version: latest
+          check-latest: true
+          cache: "yarn"
+      - name: Install
+        run: |
+          yarn install
+      - uses: actions/download-artifact@v4
+        with:
+          name: babel-artifact
+      - name: Extract artifacts
+        run: tar -xf babel-artifact.tar; rm babel-artifact.tar
+      - name: Run tests
+        run: |
+          yarn c8 jest --ci
+          yarn test:esm
+        env:
+          BABEL_ENV: test
+          BABEL_COVERAGE: true
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
This should fix the error we are currently getting when uploading the coverage report from `main`:
```
There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 429 - {"message":"Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 1621s."}
```